service/topology_state_machine.hh

```diff
@@ -200,6 +200,10 @@ struct raft_topology_snapshot {
 
     // Mutations for system.topology_requests table
     std::vector<canonical_mutation> topology_requests_mutations;
+
+    // Mutations for system_auth_v2.* tables
+    // FIXME: handle this with rpc streaming instead as we can't guarantee size bounds.
+    utils::chunked_vector<canonical_mutation> auth_mutations;
```

Comment:
I think you missed to add this field to `raft_topology_snapshot` in idl/storage_service.idl.hh. 
Was the `test_auth_raft_snapshot_transfer` test passing without it?

Reply:
oddly, yes it works. What I do:
- shutdown 1 server
- insert data
- start 1 server
- shutdown 2 other servers
- check if data is present on 1st server

And this works, perhaps there is some other mechanism involved ?

Reply:
AFAIK snapshots are only created every 1000 group 0 commands. In the aforementioned scenario, the joining node most likely received the initial snapshot and then received a log of commands that took it from the initial state to the current one.

The classic trick to force creation of another snapshot is to issue 1000 schema changes. However, @kbr-scylla recently introduced an API to force snapshot creation, perhaps you can use it in your tests: https://github.com/scylladb/scylladb/pull/16816

Reply:
so following the same logic it could also happen that your new node received older snapshot and then log of commands. or is it guaranteed in the code for new node but not for node restarts?

I'll add snapshot trigger and see what happens (expecting test to fail)

Reply:
From `raft/README.md`:

```
The snapshots are used in two ways:
- to manage Raft log length, i.e. be able to truncate
  the log when it grows too big; to truncate a log,
  the library takes a new state machine snapshot and
  erases most log entries older than the snapshot;
- to bootstrap a new member of the cluster or
  catch up a follower that has fallen too much behind
  the leader and can't use the leader's log alone; in
  this case the library instructs the user state machine
  on the leader to transfer its own snapshot (identified by snapshot
  id) to the specific follower, identified by `raft::server_id`.
  It's then the responsibility of the user state machine
  to transfer its compact state to the peer in full.
```

Snapshot transfer is guaranteed when booting a node. It's neither guaranteed nor forbidden when a node restarts; a snapshot will be transferred if a node fell behind too much.